src/client/clientsocket.rs

```diff
@@ -124,33 +196,39 @@ impl ClientSocket {
     }
     match TcpStream::connect(address).await {
       Ok(stream) => {
-        if self.debug {
-          println!(
-            "Successfully connected to server in port {}",
-            self.server_port
-          );
-        }
+        self.stream = Some(Arc::new(stream));
         if self.handshake_http().await {
-          let debug = self.debug.clone();
-          self.reader_thread = Some(thread::spawn(move || {
-            let _ = Self::reader_loop(stream, debug);
+          if self.debug {
+            println!(
+              "Successfully connected to server in port {}",
+              self.server_port
+            );
+          }
+          let (tx, rx) = channel(100);
```

Comment:
whats this number, id make a variable for it. is it going to be needed elsewhere? Maybe it could be a config or clientsocket state